integration/tests/trainium/trainium_test.go

```diff
@@ -0,0 +1,189 @@
+//go:build integration
+// +build integration
+
+package trainium
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/weaveworks/eksctl/integration/matchers"
+	. "github.com/weaveworks/eksctl/integration/runner"
+	"github.com/weaveworks/eksctl/integration/tests"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+)
+
+var (
+	defaultCluster          string
+	noInstallCluster        string
+	params                  *tests.Params
+	clusterWithNeuronPlugin string
+	clusterWithoutPlugin    string
+)
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	params = tests.NewParams("trn1")
+	defaultCluster = params.ClusterName
+	noInstallCluster = params.NewClusterName("trn1-no-plugin")
+}
+
+func TestTrainium(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+const initNG = "trn1-ng-0"
+
+var _ = BeforeSuite(func() {
+	params.KubeconfigTemp = false
+	if params.KubeconfigPath == "" {
+		wd, _ := os.Getwd()
+		f, _ := os.CreateTemp(wd, "kubeconfig-")
+		params.KubeconfigPath = f.Name()
+		params.KubeconfigTemp = true
+	}
+
+	clusterWithoutPlugin = noInstallCluster
+	clusterWithNeuronPlugin = defaultCluster
+
+	if !params.SkipCreate {
+		cmd := params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--verbose", "4",
+			"--name", clusterWithoutPlugin,
+			"--tags", "alpha.eksctl.io/description=eksctl integration test",
+			"--install-neuron-plugin=false",
+			"--nodegroup-name", initNG,
+			"--node-labels", "ng-name="+initNG,
+			"--nodes", "1",
+			"--node-type", "trn1.2xlarge",
+			"--version", params.Version,
+			"--kubeconfig", params.KubeconfigPath,
+		)
+		Expect(cmd).To(RunSuccessfully())
+
+		cmd = params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--verbose", "4",
+			"--name", clusterWithNeuronPlugin,
+			"--tags", "alpha.eksctl.io/description=eksctl integration test",
+			"--nodegroup-name", initNG,
+			"--node-labels", "ng-name="+initNG,
+			"--nodes", "1",
+			"--node-type", "trn1.xlarge",
+			"--version", params.Version,
+			"--kubeconfig", params.KubeconfigPath,
+		)
+		Expect(cmd).To(RunSuccessfully())
```

Comment:
Is this a blocker, or can I make it a fast-follow?

Reply:
Actually, looks like these tests are going to fail anyway.  Trn1 instance types are only available in a single AZ in us-west-2 and us-east-1 right now.  This will return an error unless we manually specify the zones for the cluster and node group.

Reply:
>Is this a blocker, or can I make it a fast-follow?

It's not a blocker for getting this out.

Reply:
>Actually, looks like these tests are going to fail anyway. Trn1 instance types are only available in a single AZ in us-west-2 and us-east-1 right now. This will return an error unless we manually specify the zones for the cluster and node group.

I saw the issue you opened. We should either specify the zones here or remove this integration test as we can't release with a failing integration test. I'd prefer the former. For the latter case, it'd help if we can get the output from a manual run.

Reply:
I added a method to the Trainium tests which handles the zone thing.  It's not ideal, but it ran and passed.  I will spend some time figuring out how to handle this in the core codebase and update the ticket I created earlier:

https://github.com/weaveworks/eksctl/issues/5933